### PR TITLE
TMA-712 if not run by test, the check_helper now only warns, not fails

### PR DESCRIPTION
**Pull request by Hahihula**

### CODE REVIEW

File: lib/gooddata/lcm/helpers/check_helper.rb [hunk 1]
```diff
@@ -18,20 +18,32 @@ def check_params(specification, params)
...
+                if ENV['RSPEC_ENV'] == 'test'
+                  fail "Expected parameter '#{param_name}' to be kind of '#{type}', got '#{value.class.name}'"
+                else
+                  GoodData.logger.error("Expected parameter '#{param_name}' to be kind of '#{type}', got '#{value.class.name}'")
+                end
               end
 
               if specification[param_name][:opts][:deprecated]
-                puts "WARNING: Parameter '#{param_name}' is deprecated. Please use '#{specification[param_name][:opts][:replacement]}' instead."
+                GoodData.logger.warn("WARNING: Parameter '#{param_name}' is deprecated. Please use '#{specification[param_name][:opts][:replacement]}' instead.")
```
Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'specification[param_name]' 6 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'specification[param_name][:opts]' 5 times (DuplicateMethodCall)

File: lib/gooddata/lcm/helpers/check_helper.rb [hunk 2]
```diff
@@ -18,20 +18,32 @@ def check_params(specification, params)
...
+                else
+                  GoodData.logger.error("Mandatory parameter '#{param_name}' of type '#{type}' is not specified")
+                end
               end
             else
               if type.class.const_get(:CATEGORY) == :complex && !value.is_a?(Hash)
                 puts JSON.pretty_generate(params)
-                fail "Expected parameter '#{param_name}' to be kind of '#{type}', got '#{value.class.name}'"
+                if ENV['RSPEC_ENV'] == 'test'
+                  fail "Expected parameter '#{param_name}' to be kind of '#{type}', got '#{value.class.name}'"
```
Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'value.class' 4 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'value.class.name' 2 times (DuplicateMethodCall)

File: spec/unit/helpers/check_helper_production_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      test_param_two: 'Testing param two',
+      test_param_three: 'Testing param three',
+      test_param_four: 4,
+      UPPER_case_param: 'qux'
+    }
+    GoodData::LCM2.convert_to_smart_hash(params)
+  end
+  context 'when running outside the tests' do
+    before {
+      def ENV.[](a)
```
Comment (yenkins) [inline spec/unit/helpers/check_helper_production_spec.rb]: Has the parameter name 'a' (UncommunicativeParameterName)

Comment (yenkins) [inline spec/unit/helpers/check_helper_production_spec.rb]: Has unused parameter 'a' (UnusedParameters)

Comment (yenkins) [inline spec/unit/helpers/check_helper_production_spec.rb]: Lint/UnusedMethodArgument: Unused method argument - `a`. If it's necessary, use `_` or `_a` as an argument name to indicate that it won't be used. You can also write as `[](*)` if you want the method to accept any arguments but don't care about them.

File: spec/unit/helpers/check_helper_production_spec.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    params = {
+      test_param_two: 'Testing param two',
+      test_param_three: 'Testing param three',
+      test_param_four: 4,
+      UPPER_case_param: 'qux'
+    }
+    GoodData::LCM2.convert_to_smart_hash(params)
+  end
+  context 'when running outside the tests' do
+    before {
```
Comment (yenkins) [inline spec/unit/helpers/check_helper_production_spec.rb]: Style/BlockDelimiters: Avoid using `{...}` for multi-line blocks.

File: spec/unit/helpers/check_helper_production_spec.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      def ENV.[](a)
+        return nil
```
Comment (yenkins) [inline spec/unit/helpers/check_helper_production_spec.rb]: Style/RedundantReturn: Redundant `return` detected.

File: spec/unit/helpers/check_helper_production_spec.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+      PARAMS = GoodData::LCM2::BaseAction.define_params(self) do
+        description 'Testing param four'
+        param :test_param_four, instance_of(GoodData::LCM2::Type::StringType), required: false
+      end
+      GoodData.logger = double("The Logger")
+      expect(GoodData.logger).to receive(:error)
+      expect { GoodData::LCM2::Helpers.check_params(PARAMS, params) }.not_to raise_error(/has invalid type/)
+    end
+  end
+end
```
Comment (yenkins) [inline spec/unit/helpers/check_helper_production_spec.rb]: Style/TrailingBlankLines: Final newline missing.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/61/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/873/ : FAILURE in 2m 04s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1141/ : FAILURE in 1m 50s

File: spec/unit/helpers/check_helper_production_spec.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    GoodData::LCM2.convert_to_smart_hash(params)
+  end
+  context 'when running outside the tests' do
+    let(:mocked_logger) { double(Logger) }
+    before do
+      expect(GoodData).to receive(:logger).and_return(mocked_logger)
+      expect(ENV).to receive(:[]).with('RSPEC_ENV').and_return('production')
+      expect(mocked_logger).to receive(:error)
+    end
+    it 'verifies required and writes out error message but do not fail' do
```
Comment (yenkins) [inline spec/unit/helpers/check_helper_production_spec.rb]: Similar code found in check_helper_production_spec.rb:17, check_helper_spec.rb:20 (mass = 70)

File: spec/unit/helpers/check_helper_production_spec.rb [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+      PARAMS = GoodData::LCM2::BaseAction.define_params(self) do
+        description 'Testing param one'
+        param :test_param_one, instance_of(GoodData::LCM2::Type::StringType), required: true
+      end
+      expect { GoodData::LCM2::Helpers.check_params(PARAMS, params) }.not_to raise_error(/Mandatory/)
+    end
+
+    it 'checks types and writes out error message but do not fail' do
```
Comment (yenkins) [inline spec/unit/helpers/check_helper_production_spec.rb]: Similar code found in check_helper_production_spec.rb:25, check_helper_spec.rb:40 (mass = 70)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/68/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/880/ : SUCCESS in 2m 01s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1148/ : FAILURE in 1m 58s

File: lib/gooddata/lcm/types/scalar/integer.rb [hunk 9]
```diff
@@ -13,8 +13,8 @@ class IntegerType < BaseType
         CATEGORY = :scalar
         INTERNAL_TYPE = Integer
 
-        def check(_value)
-          true
+        def check(value)
```
Comment (yenkins) [inline lib/gooddata/lcm/types/scalar/integer.rb]: Has unused parameter 'value' (UnusedParameters)

Comment (yenkins) [inline lib/gooddata/lcm/types/scalar/integer.rb]: Lint/UnusedMethodArgument: Unused method argument - `value`. If it's necessary, use `_` or `_value` as an argument name to indicate that it won't be used. You can also write as `check(*)` if you want the method to accept any arguments but don't care about them.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/70/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/882/ : FAILURE in 1m 46s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1150/ : SUCCESS in 34s

File: lib/gooddata/lcm/types/scalar/integer.rb (hunk 9 above)
Comment (yenkins) [inline lib/gooddata/lcm/types/scalar/integer.rb]: Doesn't depend on instance state (maybe move it to another class?) (UtilityFunction)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/73/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/885/ : SUCCESS in 1m 57s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1153/ : SUCCESS in 38s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/78/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/889/ : SUCCESS in 52s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1157/ : SUCCESS in 2m 02s

**Comment by kubamahnert** [comment]
ok to merge

**Pull request merged**